packages/react-ui/src/services/displays/hatchifyDisplays.tsx

```diff
@@ -292,18 +293,45 @@ export function getDisplays(
   // casting as JSX.Element because helper functions require access to
   // `child.type.name` and `child.props`
   const childArray = ReactChildren.toArray(children) as JSX.Element[]
-
-  let displays = hasValidChildren(HatchifyAttributeDisplay.name, childArray)
+  let displays = hasValidChildren(
+    HatchifyAttributeDisplay.displayName ?? "",
+    childArray,
+  )
     ? getDisplaysFromChildren(schema, defaultValueComponents, childArray)
     : getDisplaysFromSchema(
         schema,
         defaultValueComponents,
         valueComponents || null,
       )
 
-  if (hasValidChildren(HatchifyExtraDisplay.name, childArray)) {
+  if (hasValidChildren(HatchifyExtraDisplay.displayName ?? "", childArray)) {
     displays = injectExtraDisplays(displays, defaultValueComponents, childArray)
   }
 
   return displays
 }
+
+export function getEmptyList(children: React.ReactNode): () => JSX.Element {
+  const childArray = ReactChildren.toArray(children) as JSX.Element[]
+
+  const emptyListPresent = hasValidChildren(
+    HatchifyEmptyList.displayName ?? "",
+    childArray,
+  )
+
+  const emptyChild = childArray.find(
+    (child) => child.type.name === HatchifyEmptyList.displayName,
+  )
+
```

Comment:
i think this function can be cleaned up a bit, we don't need to do both `emptyListPresent` and `emptyChild`. i know I have `hasValidChildren` there for some reason that I can't remember off the top of my head 😄 but with this setup here we're looping in `hasValidChildren` over children to see if one of the children is an EmptyList, and then we're filtering it out in line #322 and repeating the processing over children.

we should be able to just do:

```js
function {
const childArray = ...

const emptyChild = ...

const emptyDisplay = emptyChild?.props.children || undefined

const EmptyList = () => emptyDisplay || <div>There are no...

return EmptyList
}
```
something like that